### ISSUE TITLE
[BUG] DSDHACKED things appearing as unknown things in GZDoom

### ISSUE THREAD

**Issue by rakohus** [ZDoom/gzdoom#2619]
### GZDoom version

g4.12.2

### Which game are you running with GZDoom?

Doom 2

### What Operating System are you using?

Windows 10

### Please describe your specific OS version

Win 10

### Relevant hardware info

N/A

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

While implementing dsdhacked support in Odamex, I was using dsda and gz as references for behavior and noticed this. I pulled these things straight out of Eviternity 2 for testing. They appear correctly in Eviternity 2 RC5 across both ports.

In the supplied wad, they only spawn in dsdadoom. In gz, they appear as the unknown <!> thing. 

### Steps to reproduce the behaviour.

Run the supplied wad.
[ev2dsd.zip](https://github.com/user-attachments/files/16120819/ev2dsd.zip)

**Comment by tomas7770** [comment]
I'm not too familiar with GZDoom's code, but from what I could figure out, the problem seems to be that `PClassActor::StaticInit` - the function that fills `PClassActor::AllActorClasses` - is called before DeHacked patches are loaded.

The new classes defined in the patch aren't added to `PClassActor::AllActorClasses`, and in turn `RegisterIDs()` isn't called for them, so they aren't added to `DoomEdMap`, preventing them from spawning in maps (though the classes exist and can be summoned through the console).

**Comment by SirBofu** [comment]
I'm currently running into this issue with my own WAD, which is going well past the 250 range and is intended to be cross-port compatible. It looks like there needs to be handling for an arbitrary number of Deh_Actor_x things, and this should probably extend to ZMAPINFO as well.

**Comment by XaserAcheron** [comment]
FYI, this ought to be treated as a high-priority issue -- several projects have had to resort to recreating their entire MBF21 patch in DECORATE/ZScript to work around this, which is a huge amount of additional effort that ought not be necessary.

**Comment by RicardoLuis0** [comment]
yeah, i just don't know much at all about the dehacked compiler (and about how it handles DSDHacked) so i haven't touched it yet, but i'll see if i can hammer out a patch for 4.14.2

**Comment by RicardoLuis0** [comment]
oh wait nvm, just saw you submitted a patch for it as well 😅
